src/Collect/Support/alias.php

```diff
@@ -1,35 +1,20 @@
 <?php
 
 $aliases = [
-    /**
-     * Contracts
-     */
-    Tightenco\Collect\Contracts\Support\Arrayable::class => Illuminate\Contracts\Support\Arrayable::class,
-    Tightenco\Collect\Contracts\Support\Htmlable::class => Illuminate\Contracts\Support\Htmlable::class,
-    Tightenco\Collect\Contracts\Support\Jsonable::class => Illuminate\Contracts\Support\Jsonable::class,
-
-    /**
-     * Support
-     */
-    Tightenco\Collect\Support\Arr::class => Illuminate\Support\Arr::class,
-    Tightenco\Collect\Support\Collection::class => Illuminate\Support\Collection::class,
-    Tightenco\Collect\Support\HigherOrderCollectionProxy::class => Illuminate\Support\HigherOrderCollectionProxy::class,
-    Tightenco\Collect\Support\HtmlString::class => Illuminate\Support\HtmlString::class,
-
-    /**
-     * Support\Debug
-     */
-    Tightenco\Collect\Support\Debug\Dumper::class => Illuminate\Support\Debug\Dumper::class,
-    Tightenco\Collect\Support\Debug\HtmlDumper::class => Illuminate\Support\Debug\HtmlDumper::class,
-
-    /**
-     * Support\Traits
-     */
-    Tightenco\Collect\Support\Traits\Macroable::class => Illuminate\Support\Traits\Macroable::class,
+    Illuminate\Support\Collection::class => Tightenco\Collect\Support\Collection::class,
+    Illuminate\Support\Arr::class => Tightenco\Collect\Support\Arr::class,
+    Illuminate\Support\Carbon::class => Tightenco\Collect\Support\Carbon::class,
+    Illuminate\Support\HigherOrderCollectionProxy::class => Tightenco\Collect\Support\HigherOrderCollectionProxy::class,
+    Illuminate\Support\HigherOrderTapProxy::class => Tightenco\Collect\Support\HigherOrderTapProxy::class,
+    Illuminate\Support\HtmlString::class => Tightenco\Collect\Support\HtmlString::class,
+    Illuminate\Support\Optional::class => Tightenco\Collect\Support\Optional::class,
+    Illuminate\Support\Str::class => Tightenco\Collect\Support\Str::class,
+    Illuminate\Support\Debug\Dumper::class => Tightenco\Collect\Support\Debug\Dumper::class,
+    Illuminate\Support\Debug\HtmlDumper::class => Tightenco\Collect\Support\Debug\HtmlDumper::class,
 ];
 
-foreach ($aliases as $tighten => $illuminate) {
-    if (! class_exists($illuminate) && ! interface_exists($illuminate) && ! trait_exists($illuminate)) {
-        class_alias($tighten, $illuminate);
+foreach ($aliases as $illuminate => $tighten) {
```

Comment:
Isnt this backwards? It's flipping from aliasing tighten to illuminate to now aliasing illuminate to tighten

Reply:
It's actually saying, if Laravel is present, ignore tightenco/collect and use it. This was due to the fact that the namespace was the same before, I think. But now, for example, a macro created using tightenco/collect will work on both namespaces, but not I'm not sure if this is even desirable, it may be a source of conflicts if the user doesn't realize that Collection is the very same all around.

Reply:
The point of the alias is:

In the past if a package developer wanted to use Collect, and then their package was pulled into Laravel with a different version of Illuminate/Support/Collect than they were expecting, they were stuck. They were refering to Illuminate/Support/Collect from Collect and were expecting it to refer to the version from their dependency but now they're getting the Laravel version (if I remember correctly.)

The goal of adding the alias is this:

If you don't care, just use the Illuminate namespace. Behavior will be predictable for 80% of use cases.

But if you do care, because you're concerned about the situation I described above, you can depend on the Tighten namespace. In that case, you'll get the Illuminate dependency (which is NOT the version you want) under Illuminate, but the Collect dependency will still offer it as a Tightenco namespace so you can still use that version.

I *think* that's the goal there.

https://github.com/tightenco/collect/issues/54

Reply:
As the alias is being autoloaded, if `Illuminate\Support\Collection` is available, the namespace `Tightenco\Collect` won't matter anymore, it was "renamed" to `Illuminate` at runtime. Even if the user forced installation of `Tightenco\Collect 5.5.0`, PHP will still refer to `Illuminate\Collection 5.5.33`, unless the package is installed on a non-Laravel app, where there's no `Illuminate\Support\Collection` available to fulfill the alias condition.

Reply:
I'm not following.

Let's walk through a few scenarios:

- MyDependency requires Tightenco/Collect 5.5.33
- MyDependency has code in it which relies on collection features that were introduced in 5.5.32, so it uses the Tightenco/Collect namespace when it uses those features.

## Scenario 1
MyApp is a Laravel App using 5.5.1
MyApp includes MyDependency

This code, which is @damiani's original code, runs as a part of the bootstrap:

```php
foreach ($aliases as $tighten => $illuminate) {
    // if (Illuminate Collection is not defined) ... which it is so no aliasing happens...
    if (! class_exists($illuminate) && ! interface_exists($illuminate) && ! trait_exists($illuminate)) {
        class_alias($tighten, $illuminate);
    }
}
```

The conditional is false because those classes *are* defined; therefore there is no aliasing. At this point, `Tightenco/Collect` is at 5.5.33 and `Illuminate/Support/Collect` is at 5.5.1.

## Scenario 2
MyApp is a Slim app
MyApp includes MyDependency

This code, which is @damiani's original code, runs as a part of the bootstrap:

```php
foreach ($aliases as $tighten => $illuminate) {
    if (! class_exists($illuminate) && ! interface_exists($illuminate) && ! trait_exists($illuminate)) {
        class_alias($tighten, $illuminate);
    }
}
```

The conditional is true because those classes *are not* defined; therefore the users can use either the Tighten or the Illuminate namespaced versions and get 5.5.33.

## Scenario 3
MyApp is a Slim App
MyApp includes Tightenco/Collect

This code, which is @damiani's original code, runs as a part of the bootstrap:

```php
foreach ($aliases as $tighten => $illuminate) {
    if (! class_exists($illuminate) && ! interface_exists($illuminate) && ! trait_exists($illuminate)) {
        class_alias($tighten, $illuminate);
    }
}
```

The conditional is true because those classes *are not* defined; therefore the users can use either the Tighten or the Illuminate namespaced versions and get 5.5.33.



Am I missing anything? Thanks!

Reply:
The alias is also there for backwards compatibility, so that people who have code depending on `tightenco/collect`, but who refer to the classes using the `Illuminate` namespace (as everyone does, currently) won't have all their code blow up when we switch to `Tightenco`. See https://github.com/tightenco/collect/issues/54#issuecomment-329594885.